lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -211,14 +211,24 @@ namespace Js
         while (true)
         {
             const PropertyRecord *currentPropertyRecord = GetTypePath()->GetPropertyIdUnchecked(currentSlotIndex);
-            currentType = currentTypeHandler->PromoteType<false>(currentType, PathTypeSuccessorKey(currentPropertyRecord->GetPropertyId(), currentAttributes), false, scriptContext, instance, &currentSlotIndex);
+            PropertyIndex index;
+            currentType = currentTypeHandler->PromoteType<false>(currentType, PathTypeSuccessorKey(currentPropertyRecord->GetPropertyId(), currentAttributes), false, scriptContext, instance, &index);
             currentTypeHandler = PathTypeHandlerBase::FromTypeHandler(currentType->GetTypeHandler());
 #if ENABLE_FIXED_FIELDS
 #ifdef SUPPORT_FIXED_FIELDS_ON_PATH_TYPES
             currentTypeHandler->InitializePath(
                 instance, currentSlotIndex, currentTypeHandler->GetPathLength(), scriptContext, [=]() { return typePath->GetIsFixedFieldAt(currentSlotIndex, currentTypeHandler->GetPathLength()); });
 #endif
 #endif
+            if (currentAttributes == ObjectSlotAttr_Setter)
+            {
+                PropertyIndex getterIndex = currentTypeHandler->GetPropertyIndex(currentPropertyRecord->GetPropertyId());
+                Assert(getterIndex != Constants::NoSlot);
+                if (currentTypeHandler->GetAttributes(getterIndex) & ObjectSlotAttr_Accessor)
```

Comment:
>currentTypeHandler->GetAttributes(getterIndex) & ObjectSlotAttr_Accessor [](start = 20, length = 72)

Wouldn't this condition be always true?

Reply:
Weirdly, no. In an object literal, for instance, you can initialize "get x", followed by a data property x, followed by "set x".